feat(dew): add an one-time resource to create a datakey without plaintext (#6996)
huaweicloud/terraform-provider-huaweicloud: docs/resources/kms_datakey_without_plaintext.md

---
subcategory: "Data Encryption Workshop (DEW)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_kms_datakey_without_plaintext"
description: |-
  Manages a KMS datakey without plaintext resource within HuaweiCloud.
---

# huaweicloud_kms_datakey_without_plaintext

Manages a KMS datakey without plaintext resource within HuaweiCloud.

-> The current resource is a one-time resource, and destroying this resource will not recover the created datakey,
but will only remove the resource information from the tfstate file.

## Example Usage

```hcl
variable "key_id" {}

> resource "huaweicloud_kms_datakey_without_plaintext" "test" {
>   key_id         = var.key_id
> }

Reviewer: ```suggestion
resource "huaweicloud_kms_datakey_without_plaintext" "test" {
  key_id = var.key_id
}
```

```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used.
  Changing this creates a new resource.

> * `key_id` - (Required, String) Specifies the key ID.

Reviewer: The parameter can update?

Author: No，modify it.

  The valid length is `36` bytes, meeting regular match **^[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}$**.
  For example: **0d0466b0-e727-4d9c-b35d-f84bb474a37f**.

* `key_spec` - (Optional, String, NonUpdatable) Specifies the generated key bit length.
  The valid values are as follows:  

>   + **AES_256**: a 256-bit symmetric key.

Reviewer 2: `256` ?

  + **AES_128**: a 128-bit symmetric key.
  
* `datakey_length` - (Optional, String, NonUpdatable) Specifies the bit length of the datakey.

>   The value must be a multiple of `8`, ranging from `8` to `8192`.

Reviewer: ```suggestion
  The value must be a multiple of `8`, the valid value ranges from `8` to `8,192`.
```

> -> For the two fields of `datakey_length` and `key_spec`,if neither of them is specified,

Reviewer: ```suggestion
-> For the two fields of `datakey_length` and `key_spec`, if neither of them is specified,
```
